libia2/include/ia2.h

```diff
@@ -44,6 +44,41 @@ static uint32_t ia2_get_pkru() {
   return pkru;
 }
 
+#define STACK_SIZE (4 * 1024 * 1024)
+
+#ifdef LIBIA2_INSECURE
+#define _IA2_DEFINE_SIGNAL_HANDLER(function, pkey)    \
+    __asm__(".global ia2_sighandler_" #function "\n"  \
+            "ia2_sighandler_" #function ":\n"         \
+            "jmp " #function "\n")
+#else
+#define _IA2_DEFINE_SIGNAL_HANDLER(function, pkey)    \
+    __asm__(".global ia2_sighandler_" #function "\n"  \
+            "ia2_sighandler_" #function ":\n"         \
+            "movq %rcx, %r10\n"                       \
```

Comment:
Not a blocker, but reading this reminded me that we don't have `endbr64` instructions here (or likely a lot of places) for CFI...

Reply:
Actually I have no idea if signal handler calls are considered indirect branches for the purposes of CFI.

Reply:
I don't _think_ so? But we'll find out, that can all be handled at once.